hyperion-explorer/src/app/app.component.ts

```diff
@@ -1,11 +1,18 @@
-import {Component} from '@angular/core';
-
+import { Component } from '@angular/core';
+import { LaunchDarklyService } from './services/launch-darkly/launch-darkly.service';
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+  styleUrls: ['./app.component.css'],
 })
 export class AppComponent {
+  constructor() {
+    this.initLaunchDarkly();
+  }
+
+  private async initLaunchDarkly(): Promise<LaunchDarklyService> {
+    const featureFlagClient = new LaunchDarklyService();
 
-  constructor() {}
+    return featureFlagClient;
```

Comment:
This return value is never used, maybe this function can just be `void`?  It will seem a little weird though to just have a function that only does `new LaunchDarklyService()` though.

Maybe in `LaunchDarklyService`, the `initLaunchDarkly` function should be public.  Then this function will look like:

```
privite initLaunchDarkly(): void {
    const featureFlagClient = new LaunchDarklyService();
    
    featureFlagClient.initLaunchDarkly();
}
```

Reply:
I would go one further and just move the logic into the constructor. There's no reason to have this a separate function when it's only 2 lines.

```suggestion
    constructor() {
        this.featureFlagClient = new LaunchDarklyService();
    }
```

Also side note as Drew mentioned this method didn't need to be async. But if it were then it can't be called in a constructor because constructors can't be async. You were missing the await in the constructor which probably masked this problem. You could still call it from the constructor if you handle the promise it returns instead of awaiting it but that would still potentially cause problems since whatever you're initializing in the constructor won't actually be done once it returns so it could throw errors on the next line of code if you try to use that thing that isn't ready yet.